sos/report/plugins/filesys.py

```diff
@@ -75,18 +72,36 @@ def setup(self):
         if self.get_option('lsof'):
             self.add_cmd_output("lsof -b +M -n -l -P", root_symlink="lsof",
                                 priority=50)
+        option_list = [
+            PluginOpt('dumpe2fs', default=False, desc='dump filesystem info'),
+            PluginOpt('frag', default=False,
+                      desc='collect filesystem fragmentation status')
+        ]
 
+    def setup(self):
         dumpe2fs_opts = '-h'
         if self.get_option('dumpe2fs'):
             dumpe2fs_opts = ''
-        mounts = '/proc/mounts'
-        ext_fs_regex = r"^(/dev/\S+).+ext[234]\s+"
-        for dev in self.do_regex_find_all(ext_fs_regex, mounts):
-            self.add_cmd_output("dumpe2fs %s %s" % (dumpe2fs_opts, dev),
-                                tags="dumpe2fs_h")
+        allfs = self.get_devices_by_fstype('ext')
+        if allfs:
+            for fs in allfs:
```

Comment:
Same here, we can drop the conditional and unnest the entire block by one level here.